### PR TITLE
Fix_status_dropdown

### PR DESCRIPTION
**Pull request by Smokashi23**
Fixed the dropdown of status based on role and accessibility

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -52,7 +53,7 @@ def load_current_user(payload)
   end
 
   def serialize_resource(resources, serializer, root = nil, extra = {} )
-    opts = { each_serializer: serializer, root: root }.merge(extra)
+    opts = { each_serializer: serializer, root: root, scope: current_user }.merge(extra)
```
Comment (Addy-tea-ya) [inline app/controllers/application_controller.rb:56]: We are using `serialize_resource` method at many places, what is the impact of adding a scope here? Please add a justification here.

File: app/serializers/ticket_serializer.rb [hunk 2]
```diff
@@ -29,8 +29,9 @@ def ask_for_update
     Ticket::ASK_FOR_UPDATE
   end
 
-  def permited_transitions
+   def permited_transitions
```
Comment (Addy-tea-ya) [inline app/serializers/ticket_serializer.rb:32]: Remove trailing space, do not push trailing spaces.

File: app/serializers/ticket_serializer.rb [hunk 3]
```diff
@@ -29,8 +29,9 @@ def ask_for_update
...
     status = object.status
+    role = scope&.role&.name
```
Comment (Addy-tea-ya) [inline app/serializers/ticket_serializer.rb:34]: Why do we need to use scopes?, ideally serializers should be independent of logged in user, you should change code calling this serializer to fit your requirement. However, as a quick fix, can we pass current user from controller as a argument only for this serializer.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@
 class ApplicationController < ActionController::API
   before_action :authenticate
   load_and_authorize_resource
+  serialization_scope :current_user
```
Comment (Addy-tea-ya) [inline app/controllers/application_controller.rb:6]: This is a base controller, adding scope here will impact everything, if you are sure about the impact, please add your justification here.

**Review by Addy-tea-ya — CHANGES_REQUESTED** [review]
I would suggest not to do changes in base files, if possible try updating the specific files you need.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Smokashi23) [reply app/controllers/application_controller.rb:56]: By adding scope here we can access it throgh serializer and i have checked the method  serialize_resource it is still functions correctly

File: app/serializers/ticket_serializer.rb (hunk 3 above)
Comment (Smokashi23) [reply app/serializers/ticket_serializer.rb:34]: i had tried the same to pass through controller but its getting null I will check again

**Review by Addy-tea-ya — APPROVED** [review]
LGTM

**Pull request merged**